addons/l10n_es_edi_facturae/__init__.py

```diff
@@ -5,35 +5,10 @@
 from . import models
 from . import wizard
 
-def _edit_tax_types(env, template_data):
-    """
-    Applies all existing tax l10n_es_edi_facturae_tax_type field to their proper value if any link between tax and their template is found
-    """
-    concerned_company_ids = [
-        company.id
-        for company in env.companies
-        if company.chart_template and company.chart_template.startswith('es_')
-    ]
-    if not concerned_company_ids:
-        return
-    current_taxes = env['account.tax'].search(env['account.tax']._check_company_domain(concerned_company_ids))
-    if not current_taxes:
-        return
-    xmlid2tax = {
-        xml_id.split('.')[1].split('_', maxsplit=1)[1]: env['account.tax'].browse(record)
-        for record, xml_id in current_taxes.get_external_id().items() if xml_id.startswith('account.')
-    }
-    for xmlid, values in template_data.items():
-        # Only update the tax_type fields
-        oldtax = xmlid2tax.get(xmlid)
-        if oldtax and oldtax.l10n_es_edi_facturae_tax_type != values.get('l10n_es_edi_facturae_tax_type'):
-            oldtax.l10n_es_edi_facturae_tax_type = values.get('l10n_es_edi_facturae_tax_type')
 
 def _l10n_es_edi_facturae_post_init_hook(env):
     """
     We need to replace the existing spanish taxes following the template so the new fields are set properly
     """
     if env['account.tax'].search_count([('country_id', '=', env.ref('base.es').id)], limit=1):
-        with file_open('l10n_es_edi_facturae/data/template/account.tax-es_common.csv') as template_file:
-            template_data = {record['id']: record['l10n_es_edi_facturae_tax_type'] for record in csv.DictReader(template_file)}
```

Comment:
@seifgneedy I suppose you just refactored for it to be called from _pre_reload_data, but any idea why we do not just do it like https://github.com/odoo/odoo/blob/b637ed70a0d67f51a8d152ebf484dd3fc9811627/addons/l10n_it_edi_withholding/__init__.py#L10

Reply:
It doesn't work in our case because the way it's done in IT withholding is adding new taxes to the already existing taxes so `ChartTemplate.load_data` will load the tax file and add the new taxes to the existing taxes in DB(with additional columns if needed).

But in our case, we are adding a column/field(`l10n_es_edi_facturae_tax_type`) to the existing loaded and stored taxes, so we are modifying existing taxes manually as it can't be done through `ChartTemplate.load_data`.

Therefore my approach here is to refactor the code to manually add the column to be used in two places:  
* post init hook, for new customers installing the module to have the taxes updated properly.
* `_pre_reload_data` is used to update existing customers' taxes when clicking the `Reload CoA` button.

Reply:
@seifgneedy What about https://github.com/odoo/enterprise/blob/b4b7ac0f8e0f4ec573da0745f8eb69fbf86b6bcc/l10n_co_edi/__init__.py#L7 then?  There is it just done for the new field. Of course, we could already merge this.